Objects/funcobject.c

```diff
@@ -162,7 +162,7 @@ PyFunction_NewWithQualName(PyObject *code, PyObject *globals, PyObject *qualname
     PyObject *consts = code_obj->co_consts;
     assert(PyTuple_Check(consts));
     PyObject *doc;
-    if (PyTuple_Size(consts) >= 1) {
+    if (PyTuple_Size(consts) >= 1 && code_obj->co_flags & CO_HAS_DOCSTRING) {
```

Comment:
If the doc flag is set and `co_consts` is empty it's a bug. I'd make it `if (doc flag)` and in the body assert that `co_consts` is not empty.